ppl/archive/import.go

```diff
@@ -377,38 +359,17 @@ func (cw *chunkWriter) closeWithTs(ctx context.Context, firstTs, lastTs nano.Ts)
 		Size:        cw.dataFileWriter.Position(),
 	}
 	if err := writeChunkMetadata(ctx, chunkMetadataPath(cw.ark, cw.tsd, cw.id), metadata); err != nil {
+		cw.seekIndex.Abort()
 		return Chunk{}, err
 	}
-	// Write the time seek index into the archive, feeding it the key/offset
-	// records written to indexTempPath.
-	tf, err := fs.Open(cw.indexTempPath)
-	if err != nil {
-		return Chunk{}, err
-	}
-	defer func() {
-		tf.Close()
-		os.Remove(tf.Name())
-	}()
-	zctx := resolver.NewContext()
-	tfr := zngio.NewReader(tf, zctx)
-	chunk := metadata.Chunk(cw.id)
-	idxURI := chunk.seekIndexPath(cw.ark)
-	idxWriter, err := microindex.NewWriter(zctx, idxURI.String(),
-		microindex.Keys("ts"),
-		microindex.FrameThresh(framesize),
-		microindex.Order(cw.ark.DataOrder),
-	)
-	if err != nil {
-		return Chunk{}, err
-	}
-	if err = zbuf.CopyWithContext(ctx, idxWriter, tfr); err != nil {
-		idxWriter.Abort()
+	if err := cw.seekIndex.Close(); err != nil {
+		cw.seekIndex.Abort()
```

Comment:
I see that the reason to call an Abort after a failed Close is that it will try to remove the partially written index file in Abort(), which we would want to do, otherwise we need to add logic to ignore a failed index read (since a partially written index file could cause reads of its data file to fail).

For simplicity sake, how about moving the close error check & Abort call inside the index builder's Close call?

This makes me think that microindex.Writer should use an iosrc.Replacer instead of an iosrc.Writer, so that the index writer can call replacer.Abort() - on S3, that will cancel the multipart upload & ensure that there's no partially written index file.

Reply:
Sure